### PR TITLE
Topic1

### PR DESCRIPTION
**Pull request by Marcosgreppi**

### CODE REVIEW

File: 01-JS-DOM-APIs/1-1.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+function showonload () {
+  let obj = document.getElementById("hidden");  
+  obj.style.display = "block";
+}
+
+function joke () {
+  let xhttp = new XMLHttpRequest();
+  xhttp.open("GET", "http://api.icndb.com/jokes/random", true);
+  xhttp.onreadystatechange = function(event) {
+    let response = JSON.parse(event.target.response);
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-1.js]: This will blow up if the request state is not DONE.

File: 01-JS-DOM-APIs/1-1.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    document.getElementById("sectionJoke").innerHTML = response.value.joke;
+    if (xhttp.readyState === XMLHttpRequest.DONE) {
+      if (xhttp.responseText.value) {
+        document.getElementById("buttonJoke").innerHTML  = xhttp.responseText.value.joke;
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-1.js]: We are you doing this? The idea was to set the joke a section not in a button

File: 01-JS-DOM-APIs/1-1.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    function reject() {
+       document.getElementById("hidden").style.backgroundColor="red";
+    };
+    requestreusable(config).then(resolve,reject);
+  };
+
+function requestreusable (object) {
+  return new Promise( function (resolve, reject) {
+    let xhttp = new XMLHttpRequest();
+    xhttp.open("GET", object.url);
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-1.js:38]: HTTP method should be in the configuration parameter... There are others methods you can use here

File: 01-JS-DOM-APIs/1-1.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      }
+    };
+    xhttp.onerror = function () {
+      reject(this.statusText);
+    };
+  })
+}
+
+function config (methodHttp, url, asyncronic){
+ this.methodHttp = methodHttp;
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-1.js:54]: Add variables to window object is not a good practice. As we talk yesterday about scopes. Please refactor this.

File: 01-JS-DOM-APIs/1-2.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    ["Esteban Quito","Ecuador", "North Korea"],
+    ["John Smith","Great Britain", "Japan"],
+    ["Federico Lopez","Spain", "Australia"],
+  ];
+  console.log(matrix[0][0]);
+  return matrix;
+}
+
+function importMatrix(matrix) {
+  console.log(matrix);
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-2.js]: Remember to remove console.log's

File: 01-JS-DOM-APIs/ex12.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html class="html" lang="">
+  <head>
+    <title>12</title>
+    <meta charset="utf-8">
+    <link rel="stylesheet" href="style1-2.css">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <style>
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/ex12.html:8]: This should go in the stylesheet

File: 01-JS-DOM-APIs/ex12.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+  <body onload="Matrix()">
+    <table id="table" class="dataTables">
+      <tr>
+        <th class="tableTH">Name</th>
+        <th class="tableTH">Departure</th>
+        <th class="tableTH">Destination</th>
+      </tr>
+    </table>
+    <button onclick="importMatrix(matrix)" id="importmatrixbutton" type="buttonimportmatrix" name="buttonimportmatrix">Import</button>
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/ex12.html]: I see several errors here:
- `buttonimportmatrix` is not a type of button. You can find type of buttons here http://www.w3schools.com/tags/att_button_type.asp
- `onclick="importMatrix(matrix)"` pass this parameter here is wrong. You should have the matrix as a variable inside the function 
- `Import` is not a good text for the button. It is very general and could be any action

File: 01-JS-DOM-APIs/ex12.html [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    <style>
+     table, th, td {
+      border: 4px solid grey;
+      text-align: center;
+    }
+</style>
+    <script src="1-2.js"> </script>
+  </head>
+
+  <body onload="Matrix()">
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/ex12.html]: Why are you doing this here?

File: 01-JS-DOM-APIs/index.html [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+    <button type="button" id="buttonReusableJoke"onclick="ReusableJoke ()">ReusableJokeConfig</button>
+    <section id="sectionReusableJoke">
+    </section>
+
+    <hr></hr>
+    <button class="button" onclick="QSearch()">QSearch</button>
+    <div>
+    </div>
+    <label for="Search"><input type="text"input type="text" name="searcher" id="searcher" value=""/>
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/index.html:29]: </label> missing

File: 01-JS-DOM-APIs/index.html [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    <section id="sectionJoke">
+    </section>
+
+    <button type="button" id="buttonReusableJoke"onclick="ReusableJoke ()">ReusableJokeConfig</button>
+    <section id="sectionReusableJoke">
+    </section>
+
+    <hr></hr>
+    <button class="button" onclick="QSearch()">QSearch</button>
+    <div>
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/index.html:27]: Why do you leave this `div` empty?

File: 01-JS-DOM-APIs/index.html [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  <body onload="setTimeout (showonload,1000)">
+    <section id="hidden">
+      Hello world.
+    </section>
+
+    <button type="button" id="buttonJoke"onclick="joke ()">Joke</button>
+    <section id="sectionJoke">
+    </section>
+
+    <button type="button" id="buttonReusableJoke"onclick="ReusableJoke ()">ReusableJokeConfig</button>
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/index.html]: Missing space between `"` and `onClick`

Comment (afiorenza) [reply 01-JS-DOM-APIs/index.html]: Also space between `ReusableJoke` and `()` is not needed

File: 01-JS-DOM-APIs/index.html [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+      <meta name="viewport" content="width=device-width, initial-scale=1">
+      <link rel="stylesheet" href="style1-1.css">
+      <script src="1-1.js"></script>
+  </head>
+
+  <body onload="setTimeout (showonload,1000)">
+    <section id="hidden">
+      Hello world.
+    </section>
+
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/index.html:16]: Tag `<section>` missing here

File: 01-JS-DOM-APIs/index.html [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    <hr></hr>
+    <button class="button" onclick="QSearch()">QSearch</button>
+    <div>
+    </div>
+    <label for="Search"><input type="text"input type="text" name="searcher" id="searcher" value=""/>
+    <button class "Button" onclick="FreeSearch(searcher.value)">Search</button>
+    <footer class="footer">
+      <ul id="list"></ul>
+    </footer>
+
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/index.html:35]: You shouldn't leave any break line here

File: 01-JS-DOM-APIs/1-2.js [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+  for(var indexTR = 0; indexTR < 3; indexTR++ ){
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-2.js]: You could use `map` function here

File: 01-JS-DOM-APIs/1-2.js [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    var trNode = document.createElement("tr");
+    for (var indexTD = 0; indexTD < 3; indexTD++) {
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-2.js]: You could use `map` function here

File: 01-JS-DOM-APIs/1-2.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+  ];
+  console.log(matrix[0][0]);
+  return matrix;
+}
+
+function importMatrix(matrix) {
+  console.log(matrix);
+
+  for(var indexTR = 0; indexTR < 3; indexTR++ ){
+    var trNode = document.createElement("tr");
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-2.js]: The guide says we should always use let instead var. Also every variable declaration should go outside loop in order to make this function more perfomant and avoid declaration several times.

File: 01-JS-DOM-APIs/style1-1.css [hunk 17]
```diff
@@ -0,0 +1,14 @@
+html,body {
+    background-color:grey;
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/style1-1.css]: Remember to leave a space after `:`

File: 01-JS-DOM-APIs/1-1.js [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+function FreeSearch (query) {
+  let objconfig = new config ("GET", "https://api.github.com/search/repositories?q="+query, true);
+  requestreusable(objconfig).then(function(response) {
+    let list = document.getElementById("list");
+    let items = JSON.parse(response).items;
+
+    items.map(function(item) {
+      var columnNode = document.createElement("li");
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-1.js]: For performance issues, the variable declarations should go outside the loop

File: 01-JS-DOM-APIs/1-1.js [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+function QSearch () {
+  let objconfig = new config ("GET", "https://api.github.com/search/repositories?q=javascript", true);
+  requestreusable(objconfig).then(function(response) {
+    let list = document.getElementById("list");
+    let items = JSON.parse(response).items;
+
+    items.map(function(item) {
+      var columnNode = document.createElement("li");
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-1.js]: For performance issues, the variable declarations should go outside the loop

File: 01-JS-DOM-APIs/1-1.js [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+function config (methodHttp, url, asyncronic){
+ this.methodHttp = methodHttp;
+ this.url = url;
+ this.asyncronic = asyncronic;
+}
+
+function QSearch () {
+  let objconfig = new config ("GET", "https://api.github.com/search/repositories?q=javascript", true);
```
Comment (afiorenza) [inline 01-JS-DOM-APIs/1-1.js:60]: Why you are calling new operator here? also assign it to a variable if in the function you are setting the configuration to the window

**Review by afiorenza — CHANGES_REQUESTED** [review]
Several changes needed

File: 01-JS-DOM-APIs/index.html (hunk 10 above)
Comment (Marcosgreppi) [reply 01-JS-DOM-APIs/index.html:27]: I left it empty so the field for the FreeSearch is not on the same line that the QSearch and the list for both gets generated under the FreeSearch

**Comment by Marcosgreppi** [comment]
Fixed everything, pushing right now

File: 01-JS-DOM-APIs/ex12.html (hunk 6 above)
Comment (Marcosgreppi) [reply 01-JS-DOM-APIs/ex12.html:8]: fixed

File: 01-JS-DOM-APIs/index.html (hunk 12 above)
Comment (Marcosgreppi) [reply 01-JS-DOM-APIs/index.html:16]: fixed

**Review by afiorenza — APPROVED** [review]

**Pull request merged**